backend/postgres/errors.go

```diff
@@ -3,9 +3,39 @@
 
 package postgres
 
-import "errors"
+import (
+	"errors"
+	"io"
+	"net"
+
+	"github.com/go-pg/pg"
+)
 
-// errors
 var (
 	ErrNotUnique = errors.New("not unique")
 )
+
+func IsErrConnectionFailed(err error) bool {
+	// go-pg has this check internally for network errors
+	if errors.Is(err, io.EOF) {
+		return true
+	}
+
+	// go-pg has this check internally for network errors
+	_, ok := err.(net.Error)
+	if ok {
+		return true
+	}
+
+	// go-pg has similar check for integrity violation issues, here we check network issues
+	pgErr, ok := err.(pg.Error)
+	if ok {
+		code := pgErr.Field('C')
+		// We check on error codes of Class 08 — Connection Exception.
+		// https://www.postgresql.org/docs/10/errcodes-appendix.html
+		if code[0:2] == "08" {
```

Comment:
This might panic if len is insufficient, but I guess this would be a valid panci

Reply:
True, but it already checks on the Postgres error type, so there is already a guarantee that you only get error codes of this format.